Ginger/GingerCoreNET/Application Models/Learn/APIModels/ApiActionConversionUtils.cs

```diff
@@ -125,53 +133,141 @@ private void ConvertAction(bool parameterizeRequestBody, bool pullValidations, A
         {
             AutoMapper.MapperConfiguration mapConfigUIElement = new AutoMapper.MapperConfiguration(cfg => { cfg.CreateMap<Act, ActWebAPIModel>(); });
             ActWebAPIModel newActModel = mapConfigUIElement.CreateMapper().Map<Act, ActWebAPIModel>(act);
             newActModel.APImodelGUID = applicationModel.Guid;
             newActModel.Active = true;
-            newActModel.ActAppModelParameters = applicationModel.AppModelParameters;
+            newActModel.Reset();
+            newActModel.APIModelParamsValue = GetAPIActionParams(applicationModel.AppModelParameters, optionalValuesPulledFromConvertedAction);
+            //newActModel.ReturnValues = GetAPIModelActionReturnValues(applicationModel.ReturnValues); //For now we want to keep original validations on action
             return newActModel;
         }
 
+        private ObservableList<EnhancedActInputValue> GetAPIActionParams(ObservableList<AppModelParameter> paramsList, Dictionary<System.Tuple<string, string>, List<string>> optionalValuesPulledFromConvertedAction)
+        {
+            ObservableList<EnhancedActInputValue> enhancedParamsList = new ObservableList<EnhancedActInputValue>();
+            foreach (AppModelParameter apiModelParam in paramsList)
+            {
+                if (apiModelParam.RequiredAsInput == true)
+                {
+                    EnhancedActInputValue actAPIModelParam = new EnhancedActInputValue();
+                    actAPIModelParam.ParamGuid = apiModelParam.Guid;
+                    actAPIModelParam.Param = apiModelParam.PlaceHolder;
+                    actAPIModelParam.Description = apiModelParam.Description;                    
+                    foreach (OptionalValue optionalValue in apiModelParam.OptionalValuesList)
+                    {
+                        actAPIModelParam.OptionalValues.Add(optionalValue.Value);
+                    }
+
+                    //set value came from Action 
+                    foreach (System.Tuple<string, string> key in optionalValuesPulledFromConvertedAction.Keys)
+                    {
+                        if (key.Item2 == apiModelParam.Path)
+                        {
+                            if (optionalValuesPulledFromConvertedAction[key] != null && optionalValuesPulledFromConvertedAction[key].Count > 0)
+                            {
+                                actAPIModelParam.Value = optionalValuesPulledFromConvertedAction[key][0];
+                                break;
+                            }
+                        }
+                    }
+
+                    if (string.IsNullOrEmpty(actAPIModelParam.Value))
+                    {
+                        //set defualt value
+                        OptionalValue ov = apiModelParam.OptionalValuesList.Where(x => x.IsDefault == true).FirstOrDefault();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/app/Ginger-Automation/Ginger/pullRequest?prid=4513408)